### PR TITLE
Create page on session management in Pharo

### PR DESCRIPTION
**Pull request by jecisc**

### CODE REVIEW

File: General/SessionsManagement.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
+# Sessions Management
+
+ * [Description](#description)
+ * [Quick start](#quick-start)
+ * [Register a class to the session manager](#register-a-class-to-the-session-manager)
+ * [Start-up, Shut-down and Saving actions](#start-up--shut-down-and-saving-actions)
+
+## Description
+
+Pharo images have a concept of **Session**. A session start when we open a Pharo image and stop when we close it.
```
Comment (juliendelplanque) [inline General/SessionsManagement.md]:
```suggestion
Pharo images have a concept of **Session**. A session starts when one open a Pharo image and stop when one close it.
```

File: General/SessionsManagement.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+It is possible to hook your project to those sessions to do action such as executing code at startup, shutdown, image saving...
```
Comment (juliendelplanque) [inline General/SessionsManagement.md]:
```suggestion
It is possible to hook your project to those sessions to perform actions such as executing code at startup, shutdown, image saving...
```

File: General/SessionsManagement.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+## Quick start
+
+Here is a quick start to show how to set a class variable to the home folder of a user at startup and removing this value at shut down. 
+
+You need to do those actions in a `#startUp:` and `#shutDown:` present in the class side of your class.
```
Comment (juliendelplanque) [inline General/SessionsManagement.md]:
```suggestion
You need to do those actions in a `#startUp:` method and `#shutDown:` method on the class side of your class. 
```

File: General/SessionsManagement.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+			ifTrue: [ MyVariable := FileLocator home asFileReference ]
+```
+
+```Smalltalk
+MyClass class>>shutDown: isImageQuitting
+		isImageQuitting
+			ifTrue: [ MyVarible := nil ]
+```
+
+Then you need to register the class in the `SessionManager` of Pharo. This is usually done in the `#initialize` method.
```
Comment (juliendelplanque) [inline General/SessionsManagement.md]:
```suggestion
Then you need to register the class in the `SessionManager` of Pharo. This is usually done in the `#initialize` method (class-side).
```

File: General/SessionsManagement.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+## Register a class to the session manager
+
+It is possible to register classes to the `SessionManager` of Pharo in order to be able to execute code at start-up, shut-down or when we save a Pharo image.
+
+The simplest way to do that is to add this in the class initialization method of the class you want to register.
+
+```Smalltalk
+MyClass class>>initialize
+SessionManager default registerUserClassNamed: self name
```
Comment (juliendelplanque) [inline General/SessionsManagement.md]:
```suggestion
    SessionManager default registerUserClassNamed: self name
```

File: General/SessionsManagement.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+```
+  
+Sometimes, the order of execution of start-up/shut-down actions might matter. To manage this it is possible to set a priority.
```
Comment (juliendelplanque) [inline General/SessionsManagement.md]:
```suggestion
Sometimes, the order of execution of start-up/shut-down actions might matter. To manage this, it is possible to set a priority.
```

File: General/SessionsManagement.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+  
+```Smalltalk
+MyClass class>>initialize
+SessionManager default registerUserClassNamed: self name atPriority: 60
+```
+  
+The actions registered using `#registerUserClassNamed:[atPriority:]` will have a low priority in the startup/shutdown list.
+It is possible to further refine the priorities using categories. 
+
+The SessionManager has categories and the `User` one is one of them. Categories have priority between each others.
```
Comment (juliendelplanque) [inline General/SessionsManagement.md]:
```suggestion
The SessionManager has categories. `User` category one is one of them. Categories have priority between each others.
```

File: General/SessionsManagement.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+- User (`#registerUserClassNamed:[atPriority:]`)
+
+It is possible to create a new category:
+
+```Smalltalk
+SessionManager default createCategory: aCategoryName.
+SessionManager default createCategory: aCategoryName after: anotherCategoryName.
+```
+
+Then you can register classes in you new custom category:
```
Comment (juliendelplanque) [inline General/SessionsManagement.md]:
```suggestion
Then one can register classes in you new custom category:
```

File: General/SessionsManagement.md [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+Example:
+
+```Smalltalk
+MyClass class>>shutDown: isImageQuitting
+	Transcript traceCr:
+		(isImageQuitting
+			ifTrue: [ 'Image is quitting' ]
+			ifFalse: [ 'Image is been saved' ])
+```
+
```
Comment (juliendelplanque) [inline General/SessionsManagement.md]: Remove trailing empty lines.

**Comment by jecisc** [comment]
Add: 
- Warning to not register before the GUI/System/Network except in exceptional cases
- Say that by default it users should register in "User"
- Explain categories: System: init everything related to the system (FFI, File, OSEnvironment...), Network: Zinc, Git..., GUI: Events, sensors, Morphic..., Tools: IDE Pharo, Users: What you should use in most cases

**Comment by jecisc** [comment]
Duplication of content in the Quick start is wanted

**Pull request merged**